cdk/lib/support-workers.ts

```diff
@@ -0,0 +1,236 @@
+      resources: props.eventBusArns,
+    });
+    const cloudWatchLoggingPolicy = new PolicyStatement({
+      actions: [
+        "logs:Create*",
+        "logs:PutLogEvents",
+        "logs:DescribeLogStreams",
+        "cloudwatch:putMetricData",
+      ],
+      resources: ["*"],
+    });
+    const promotionsDynamoTablePolicy = new PolicyStatement({
+      actions: [
+        "dynamodb:GetItem",
+        "dynamodb:Scan",
+        "dynamodb:Query",
+        "dynamodb:DescribeTable",
+      ],
+      resources: props.promotionsDynamoTables,
+    });
+    const supporterProductDataTablePolicy = new PolicyStatement({
+      actions: ["dynamodb:PutItem", "dynamodb:UpdateItem"],
+      resources: props.supporterProductDataTables.map((table) =>
+        Fn.importValue(table)
+      ),
+    });
+
+    const createLambda = (
+      lambdaName: string,
+      additionalPolicies: PolicyStatement[] = []
+    ) => {
+      const lambdaId = `${lambdaName}Lambda`;
+      const lambda = new GuLambdaFunction(this, lambdaId, {
+        ...lambdaDefaultConfig,
+        handler: `com.gu.support.workers.lambdas.${lambdaName}::handleRequest`,
+        functionName: `${this.stack}-${lambdaName}Lambda-${this.stage}`,
+        environment: {
+          ...lambdaDefaultConfig.environment,
+        },
+        initialPolicy: [
+          s3Policy,
+          cloudWatchLoggingPolicy,
+          ...additionalPolicies,
+        ],
+      });
+      this.overrideLogicalId(lambda, {
+        logicalId: lambdaId,
+        reason: "Moving existing lambda to CDK",
+      });
+      return new LambdaInvoke(this, lambdaName, {
+        lambdaFunction: lambda,
+        outputPath: "$.Payload", // Without this, LambdaInvoke wraps the output state as described here: https://github.com/aws/aws-cdk/issues/29473
+      })
+        .addRetry({
+          errors: ["com.gu.support.workers.exceptions.RetryNone"],
+          maxAttempts: 0,
+        })
+        .addRetry({
+          errors: ["com.gu.support.workers.exceptions.RetryLimited"],
+          maxAttempts: 5,
+          interval: Duration.seconds(1),
+          backoffRate: 10, // Retry after 1 sec, 10 sec, 100 sec, 16 mins and 2 hours 46 mins
+        })
+        .addRetry({
+          errors: ["com.gu.support.workers.exceptions.RetryUnlimited"],
+          maxAttempts: 999999, //If we don't provide a value here it defaults to 3
+          interval: Duration.seconds(1),
+          backoffRate: 2,
+        })
+        .addRetry({
+          errors: [Errors.ALL], // Wildcard to capture any error which originates from outside of our code (e.g. an exception from AWS)
+          maxAttempts: 999999,
+          interval: Duration.seconds(3),
+          backoffRate: 2,
+        });
+    };
+
+    const checkoutFailure = new Pass(this, "CheckoutFailure"); // This is a failed execution we don't want to alert on
+    const failState = new Fail(this, "FailState"); // This is a failed execution we do want to alert on
```

Comment:
I assume the alarms are still going to be defined in the existing CFN for now

Reply:
Great catch, I'd missed the alarms off. Will add them.